ambry-frontend/src/main/java/com/github/ambry/frontend/GetBlobHandler.java

```diff
@@ -340,6 +341,9 @@ private String getDatasetVersion(RestRequest restRequest) throws RestServiceExce
         restResponseChannel.setHeader(RestUtils.Headers.TARGET_CONTAINER_NAME, containerName);
         restResponseChannel.setHeader(RestUtils.Headers.TARGET_DATASET_NAME, datasetName);
         restResponseChannel.setHeader(RestUtils.Headers.TARGET_DATASET_VERSION, datasetVersionRecord.getVersion());
+        restResponseChannel.setHeader(RestUtils.Headers.DATASET_EXPIRATION_TIME,
+            datasetVersionRecord.getExpirationTimeMs() == -1 ? "Infinite_Time"
+                : new Date(datasetVersionRecord.getExpirationTimeMs()));
```

Comment:
standard http response header "Expires" can show the expiration date/time for an entity. I think it would be better to use standard http header here. If this dataset version has no ttl, we don't have to set this header. but it does, we can set "Expires" header.

Reply:
I can do that, but I checked we have this setCacheHeaders in AmbrySecurityService which will always set the EXPIRES to DATE, I'm kind of confused about why we always set that for a get call.
restResponseChannel.setHeader(RestUtils.Headers.EXPIRES, restResponseChannel.getHeader(RestUtils.Headers.DATE));

Reply:
oh, my bad, EXPIRES is for browser cache, not for the blob itself. I don't think EXPIRES is going to work. :(